src/field_theory/splitting_field.lean

```diff
@@ -208,6 +229,75 @@ end
 
 end splits
 
+end polynomial
+
+
+section embeddings
+
+variables (F : Type*) [field F]
+
+/-- The bottom subalgebra is isomorphic to the field. -/
+def alg_equiv.of_bot (R : Type*) [semiring R] [nontrivial R] [algebra F R] :
+  (⊥ : subalgebra F R) ≃ₐ[F] F :=
+alg_equiv.symm $ alg_equiv.of_bijective (algebra.of_id F _)
+⟨ring_hom.injective _, λ ⟨y, hy⟩, let ⟨x, hx⟩ := algebra.mem_bot.1 hy in ⟨x, subtype.eq hx⟩⟩
+
+/-- If `p` is the minimal polynomial of `a` over `F` then `F[a] ≃ₐ[F] F[x]/(p)` -/
+def alg_equiv.adjoin_singleton {R : Type*} [comm_ring R] [algebra F R]
+  (x : R) (hx : is_integral F x) :
+  algebra.adjoin F ({x} : set R) ≃ₐ[F] adjoin_root (minimal_polynomial hx) :=
+alg_equiv.symm $ alg_equiv.of_bijective
```

Comment:
I'm not sure about the name: there is also a `adjoin_singleton_eq_range`. I'd like to have this name specify what `adjoin_singleton` is equivalent to.

Reply:
done